### PR TITLE
Replace clearColorFilter with setColorFilter for security icon

### PR DESCRIPTION
**Pull request by ionutbedregeanu**
For #11545

Clear color filter doesn't work on Api 21, 22
Even if we invalidate the view after clearColorFilter the filter is not removed, so to fix the bug we need to use setColorFilter with color Color.Transparent

### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/main/docs/changelog.md) or does not need one
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/main/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- **Breaking Changes**: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### CODE REVIEW

File: components/browser/toolbar/src/main/java/mozilla/components/browser/toolbar/display/DisplayToolbar.kt [hunk 1]
```diff
@@ -500,12 +499,7 @@ class DisplayToolbar internal constructor(
             Toolbar.SiteSecurity.INSECURE -> colors.securityIconInsecure
             Toolbar.SiteSecurity.SECURE -> colors.securityIconSecure
         }
-        if (color == Color.TRANSPARENT) {
-            views.securityIndicator.clearColorFilter()
```
Comment (Mugurell) [inline components/browser/toolbar/src/main/java/mozilla/components/browser/toolbar/display/DisplayToolbar.kt:504]: Could we limit this change to only the affected Android versions?
Also, if we know the api is broken and is used in multiple places in this file we should refactor all the occurrences.

Comment (Mugurell) [reply components/browser/toolbar/src/main/java/mozilla/components/browser/toolbar/display/DisplayToolbar.kt:504]: Opened https://github.com/mozilla-mobile/android-components/issues/11549 to investigate the other uses of this API.

File: docs/changelog.md [hunk 2]
```diff
@@ -20,9 +23,9 @@ permalink: /changelog/
 * **feature-top-sites**
   * ⚠️ **This is a breaking change**: The existing data class `TopSite` has been converted into a sealed class. [#11483](https://github.com/mozilla-mobile/android-components/issues/11483)
   * Extend `DefaultTopSitesStorage` to accept a `TopSitesProvider` for fetching top sites. [#11483](https://github.com/mozilla-mobile/android-components/issues/11483)
-  
+
```
Comment (ionutbedregeanu) [inline docs/changelog.md:26]: I'm not sure what happened here, it appears after a rebase, but if i'm trying to revert this change locally and commit it says that this file is identical with the main branch version
